app/flatpak-builtins-build-init.c

```diff
@@ -292,8 +293,15 @@ flatpak_builtin_build_init (int argc, char **argv, GCancellable *cancellable, GE
         return FALSE;
     }
 
+  g_autoptr(FlatpakRef) ref = flatpak_ref_parse (sdk_ref, NULL);
```

Comment:
Please put all variable definitions at top, especially ones with autoptr, i've seen some issues with them being freed before set otherwise. Also, creating a full FlatpakRef object is massive overkill. You should be fine with something like:
``` 
   g_auto(GStrv) sdk_ref_parts = NULL;
   sdk_ref_parts = flatpak_decompose_ref (sdk_ref, error);
```